filter/etc/e-smith/templates/etc/rspamd/local.d/actions.conf/10base

```diff
@@ -4,18 +4,20 @@
 {
 my $reject  = $rspamd{SpamKillLevel} || '15';
 my $add     = $rspamd{SpamTag2Level} || '6';
-my $rewrite = $rspamd{SpamTag2Level} || '6';
+my $rewrite = sprintf("%f", $add) + 0.01;
```

Comment:
As I understand from Perl docs, it should be

    sprintf("%.2f", $add + 0.01);

That means

- add decimal precision
- perform sum before string conversion

Reply:
`9.110000` is what we obtain, not sure if it is good for you, normally perl takes care alone to understand the value of the variable @DavidePrincipi

Reply:
The sum after sprintf makes sprintf useless!

Reply:
Originally I believed that you just want to be sure that the value is numerical and that we avoid a scientific notation....with my use of sprintf we obtain `9.11`

I must admit I'm discovering sprintf, never used before

Reply:
If you get string "9.11" we probably do not need sprintf at all. Anyway sprintf gives the full control of float conversion to string. As you see it is not trivial and many hidden rules apply. I prefer to program it explicitly, instead of relying on the language rules that I might overlook or ignore.

Reply:
A sprintf equivalent is available in almost any programming language. This is why I prefer it

Reply:
ok so we let as `sprintf("%f", $add) + 0.01`or back to `$add + 0.01`....I'm a bit lost :)

Reply:
To recap

- `$add + 0.01` gives a float: I don't know what Perl does when coerces float to string during string concatenation or interpolation
- `sprintf("%f", $add) + 0.01` is similar to the code above but with more code. I think it's a bad code
- `sprintf("%.2f", $add + 0.01)` sums two floats and gives back a string, converting the float to decimal notation, 2 digits of precision. It's exactly what I want to give to rspamd config parser.

Reply:
To recap

- `$add + 0.01` gives a float: I don't know what Perl does when coerces float to string during string concatenation or interpolation
- `sprintf("%f", $add) + 0.01` is similar to the code above but with more code. I think it's a bad code
- `sprintf("%.2f", $add + 0.01)` sums two floats and gives back a string, converting the float to decimal notation, 2 digits of precision. It's exactly what I want to give to rspamd config parser.

Reply:
http://floating-point-gui.de/languages/perl/ I read a bit on this topic